### PR TITLE
#56 - add notification tools

### LINKED ISSUE #56: Add notifications tools

### PR DESCRIPTION
**Pull request by JakubKermes**
This should close #56.

### CODE REVIEW

**Comment by JakubKermes** [comment]
Works when Notification is sent from test, not when app sends notification. Work in progress

**Comment by EwelinaSkrzypacz** [comment]
Conflicts :(

File: src/Features/Traits/Notification.php [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+    /**
+     * @Then :notification notification was sent
+     * @Then :notification notification was sent :count times
+     */
+    public function assertNotificationSent(string $notification, int $count = 1): void
+    {
+        $notificationClass = RecognizeClassHelper::recognizeObjectClass($notification);
+        $this->notificationFake->assertSentTimes($notificationClass, $count);
+    }
```
Comment (EwelinaSkrzypacz) [inline src/Features/Traits/Notification.php:48]: I think that there is missing method that checks that notification was not sent

**Review by EwelinaSkrzypacz — APPROVED** [review]

File: src/Features/Traits/Notification.php [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+    public NotificationFake $notificationFake;
+
+    /**
+     * @Given notifications are being faked
+     * @throws BindingResolutionException
+     */
+    public function fakeNotifications(): void
+    {
+        $this->notificationFake = $this->getContainer()->make(NotificationFake::class);
+        $this->getContainer()->instance(ChannelManager::class, $this->notificationFake);
```
Comment (krzysztofrewak) [inline src/Features/Traits/Notification.php:25]: Please use `\Illuminate\Contracts\Notifications\Dispatcher` contract instead of `ChannelManager`.

**Review by krzysztofrewak — CHANGES_REQUESTED** [review]

**Review by krzysztofrewak — APPROVED** [review]

**Pull request merged**